src/hooks/useVenueAccess.ts

```diff
@@ -1,26 +1,31 @@
-import { useEffect } from "react";
+import { useCallback, useEffect } from "react";
 import firebase from "firebase/app";
 
 import { AnyVenue } from "types/Firestore";
 
 import { WithId } from "utils/id";
-import { accessTokenKey } from "utils/localStorage";
-import { isTruthy, notEmpty } from "utils/types";
-import { checkAccess, getLocalStorageToken } from "functions/auth";
+import {
+  accessTokenKey,
+  checkAccess,
+  getLocalStorageToken,
+} from "utils/localStorage";
+import { isTruthy } from "utils/types";
 
-export const useVenueAccessToken = (
+export const useVenueAccess = (
   venue?: WithId<AnyVenue>,
   onDenyAccess?: () => void
 ) => {
-  useEffect(() => {
+  const denyAccess = useCallback(() => {
     if (!venue) return;
 
-    const denyAccess = () => {
-      localStorage.removeItem(accessTokenKey(venue.id));
-      onDenyAccess && onDenyAccess();
-    };
+    localStorage.removeItem(accessTokenKey(venue.id));
+    onDenyAccess && onDenyAccess();
+  }, [onDenyAccess, venue]);
+
+  useEffect(() => {
+    if (!venue) return;
 
-    if (notEmpty(venue.access)) {
+    if (venue.access) {
       console.log("venue.access", venue.access);
       const token = getLocalStorageToken(venue.id) ?? undefined;
       console.log("found token:", token);
```

Comment:
These `console.log`'s will want to be removed before this gets merged.

Reply:
I believe my last review comments also talked about how we could probably use `async`/`await` here to clean up/un-nest this `.then()` chain here as well?